docs/handling-files.rst

```diff
@@ -0,0 +1,86 @@
+.. _handling-files:
+
+Handling Files
+================
+
+.. currentmodule:: click
+
+Click has built in features to support file and file path handling. Since files and files paths are recommended to be passed in as arguments, the examples use arguments.
+
+.. _file-args:
+
+File Arguments
+-----------------
+
+Click supports working with files with the :class:`File` type. Some notable features are:
+
+*   Support for   ``-`` to mean a special file that refers to stdin/stdout. This is common pattern for POSIX command line utilities, for example wget.
+*   Deals with Unicode and bytes correctly for all versions of Python.
+
+Example:
+
+.. click:example::
+
+    @click.command()
+    @click.argument('input', type=click.File('rb'))
+    @click.argument('output', type=click.File('wb'))
+    def inout(input, output):
+        """Copy contents of INPUT to OUTPUT."""
+        while True:
+            chunk = input.read(1024)
+            if not chunk:
+                break
+            output.write(chunk)
+
+And what it does:
```

Comment:
Can we use `And from the command line:` instead of `And what it does:`? I feel like the original is better.

Reply:
Yes. That was actually an unintentional to propagate it everywhere.